test/e2e/minimal_configuration.go

```diff
@@ -195,9 +209,22 @@ var _ = Describe("Create providers with minimal specified configuration", func()
 			}
 			return false
 		}, timeout).Should(Equal(true))
+
+		Expect(k8sclient.Delete(ctx, cpProvider)).To(Succeed())
+
+		By("Waiting for the control plane provider deployment to be deleted")
+		Eventually(func() bool {
+			deployment := &appsv1.Deployment{}
+			key := client.ObjectKey{Namespace: operatorNamespace, Name: cpProviderDeploymentName}
+			isReady, err := waitForObjectToBeDeleted(k8sclient, ctx, key, deployment)
+			if err != nil {
+				return false
+			}
+			return isReady
```

Comment:
how about:
```suggestion
			isCPProviderDeleted, err := waitForObjectToBeDeleted(k8sclient, ctx, key, deployment)
			if err != nil {
				return false
			}
			return isCPProviderDeleted
```
instead?